src/PropertyList/PropertyList.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { ReactNode } from 'react';
+import Immutable from 'immutable';
+import { AttributeGrid, AttributeList, Caption, Content } from './propertlyListStyles';
+import { ThemeProvider } from '@emotion/react';
+import theme from 'src/styles/theme';
+
+export interface Property {
+    /** This is to set the caption for the PropertyList */
+    caption: string;
+    /** This is to set the caption style that needs to be added to the caption */
+    captionStyles?: string | null;
+    /** This is to set the content for the corresponding caption */
+    content: (data: any) => ReactNode;
+    /** This is to set the style for the content associated to the caption */
+    contentStyles?: string | null;
+}
+
+interface BaseProps {
+    /** This is to set the list of the property- caption, captionStyle, content and contentStyles */
+    attributes: Immutable.List<Property>;
+    /** This is to set the external class on the Property List. */
+    className?: string;
+    /** This is to set the data for the content defined in the Property */
+    data: any;
+}
+
+export interface PropertyListProps extends BaseProps {
+    /** This is to set the alignment of the content. Set variant as 'left-align' for aligning the content to left. By default it is right aligned.*/
+    variant: string;
+}
+
+/** The PropertyList Component is used to display the list of captions and their corresponding data. The default behavior of the content is right align but you can choose to
```

Comment:
I think this should be clarified to say that the default behavior is to "right-align the captions" (not the content) -- in order to demonstrate this, it might help to have a storybook example with a very long content line.

Reply:
(And variable-length captions -- that's actually where it's most obvious.)